lib/action_subscriber/bunny/subscriber.rb

```diff
@@ -43,11 +41,14 @@ def start_subscriber_for_subscription(subscription)
         if ::ActionSubscriber.configuration.resubscribe_on_consumer_cancellation
           # Add cancellation callback to rebuild subscriber on cancel.
           consumer.on_cancellation do
-            ::ActionSubscriber.logger.warn "Cancelation received for queue consumer: #{queue.name}, rebuilding subscription..."
-            bunny_consumers.delete(consumer)
-            channel.close
-            queue = subscription[:queue] = setup_queue(route)
-            start_subscriber_for_subscription(subscription)
+            properties = {
+              :consumer => consumer,
+              :consumers => bunny_consumers,
+              :route_set => self,
+              :subscription => subscription
+            }
+            env = ::ActionSubscriber::Middleware::ResubscribeEnv.new(properties)
```

Comment:
Why create a second middleware stack? Is this something we can add on a per-app basis (custom middleware)?